crates/rust-client/src/store/mod.rs

```diff
@@ -347,6 +347,12 @@ pub enum TransactionFilter {
     Uncomitted,
     /// Return a list of the transaction that matches the provided [`TransactionId`]s.
     Ids(Vec<TransactionId>),
+    /// Return a list of the expired pending transactions until the provided block number.
```

Comment:
nit: Maybe "pending" here is redundant. A committed transaction would never be expired. Maybe something like this could be more clear?
```suggestion
    /// Return a list of the expired transactions that were executed before the provided [`BlockNumber`].
```

Also, should we rename the variant to better reflect this? Maybe something like `ExpiredBefore(BlockNumber)`?